eth/downloader/downloader.go

```diff
@@ -1820,3 +1831,46 @@ func (d *Downloader) readHeaderRange(last *types.Header, count int) []*types.Hea
 	}
 	return headers
 }
+
+// reportSnapSyncProgress calculates various status reports and provides it to the user.
+func (d *Downloader) reportSnapSyncProgress(force bool) {
+	// Don't report all the events, just occasionally
+	if !force && time.Since(d.syncLogTime) < 8*time.Second {
+		return
+	}
+	// Don't report anything until we have a meaningful progress
+	var (
+		headerBytes, _  = d.stateDB.AncientSize(rawdb.ChainFreezerHeaderTable)
+		bodyBytes, _    = d.stateDB.AncientSize(rawdb.ChainFreezerBodiesTable)
+		receiptBytes, _ = d.stateDB.AncientSize(rawdb.ChainFreezerReceiptTable)
```

Comment:
This is the compressed size, so won't match up with what poeple see on charts, but I guess we can live with that

Reply:
I'd argue it will match up, since charts usually just spit out geth's disk size without caring what's there. For uncompressed, things would get a lot more complicated since we'd need to track it in a new stats on disk. Not really sure it's worth it.

Reply:
I meant "ingress charts" :) 
Anyway, yeah, definitely not worth it to try and track the "real" value, LGTM

Reply:
devp2p is also snappy encoded tho :)